### ISSUE TITLE
open in visual studio

### ISSUE THREAD

**Issue by eshagalawatta95** [dzzie/VS_LIBEMU#6]
Dear sir,I have these issues,How can I fix it? I open in vs 17 
![image](https://user-images.githubusercontent.com/54389992/82021206-1f2df880-96a8-11ea-8e97-aed52812922f.png)

**Comment by dzzie** [comment]
Hi, for some reason I did not receive notification of this. I just tested it out with VS2019 and was able to get it to compile with a few minor changes.

First to take care of the hash_map depreciation:

C/C++ -> Preprocessor -> Preprocessor Definitions 
Then Add the text ";_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS" to end minus quotes

Try to compile again. In VS2019 i then received  a couple errors because of it now not accepting uint32_t types for LPDWORD, PULONG, or DWORD*. Click on each one of these errors to goto the source line, then just add the appropriate cast such as (DWORD*)

Example:
--------------------
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2664	'DWORD GetFileSize(HANDLE,LPDWORD)': cannot convert argument 2 from 'uint32_t *' to 'LPDWORD'	scdbg	D:\_code\libemu\VS_LIBEMU_VS2019\scdbg\userhooks.cpp	3114	

Fix:
ret_val = GetFileSize( (HANDLE)hFile, (LPDWORD)&sizeHigh) + opts.adjust_getfsize;

Example:
---------------------
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2664	'BOOL CryptGetHashParam(HCRYPTHASH,DWORD,BYTE *,DWORD *,DWORD)': cannot convert argument 4 from 'uint32_t *' to 'DWORD *'	scdbg	D:\_code\libemu\VS_LIBEMU_VS2019\scdbg\userhooks.cpp	3472	

Fix:
rv = (uint32_t)CryptGetHashParam(hHash,dwParam,myData, (DWORD*)&myDataLen,dwFlags); 

Updates should take less than 5 minutes.

I still use VS2008 so will leave this to readers. I have attached a small patch file which may help.

[0001-vs2019.patch.txt](https://github.com/dzzie/VS_LIBEMU/files/4908478/0001-vs2019.patch.txt)
